extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQCompactionRunner.java

```diff
@@ -469,6 +526,8 @@ private Map<String, Object> createMSQTaskContext(CompactionTask compactionTask,
     }
     // Similar to compaction using the native engine, don't finalize aggregations.
     context.putIfAbsent(MultiStageQueryContext.CTX_FINALIZE_AGGREGATIONS, false);
+    // Add appropriate finalization to native query context.
+    context.put(QueryContexts.FINALIZE_KEY, false);
```

Comment:
What is the effect of adding this? Maybe update the comment to explain why this line is needed.

Reply:
Updated the comments.